### PR TITLE
Added an example web app todo list for demonstration

### PR DESCRIPTION
**Pull request by powerfooI**
The todo list web app will substitute OCP in the [Real World Example](https://oceanbase.github.io/ob-operator/docs/manual/appendix/example) docs.

image

### CODE REVIEW

File: distribution/oceanbase-todo/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# OceanBase Todo List
+
+OceanBase Todo List, an extremely simple todo list web application that shows how to use ob-operator.
```
Comment (chris-sun-star) [inline distribution/oceanbase-todo/README.md:3]: This not not a demo for ob-operator but an application using OceanBase as backend

File: distribution/oceanbase-todo/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+```bash
+# Build frontend app first
+cd ui
+yarn # or npm install
+yarn build # or npm run build
+
+# Run backend app
+cd ..
+go mod tidy
+DB_HOST=xxx DB_PORT=xxx DB_USER=xxx DB_PASSWORD=xxx DB_DATABASE=xxx go run main.go
```
Comment (chris-sun-star) [inline distribution/oceanbase-todo/README.md:28]: it's better to use compiled binary in README

File: distribution/oceanbase-todo/internal/dsn.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+		prefix = strings.Join(prefixes, "_") + "_"
+	}
+	dsn := DSN{
+		Host:     os.Getenv(prefix + "DB_HOST"),
+		Port:     os.Getenv(prefix + "DB_PORT"),
+		User:     os.Getenv(prefix + "DB_USER"),
+		Password: os.Getenv(prefix + "DB_PASSWORD"),
+		Database: os.Getenv(prefix + "DB_DATABASE"),
+	}
+	if dsn.Port == "" {
```
Comment (chris-sun-star) [inline distribution/oceanbase-todo/internal/dsn.go:33]: if the port is not provided, an error should be returned, DSN should be a low level model, and it's not necessary to add OceanBase related logic

File: distribution/oceanbase-todo/ui/README.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
+# React + TypeScript + Vite
```
Comment (chris-sun-star) [inline distribution/oceanbase-todo/ui/README.md:1]: This seems like an auto-generated readme file, is it necessary to change it

File: distribution/oceanbase-todo/README.md (hunk 2 above)
Comment (powerfooI) [reply distribution/oceanbase-todo/README.md:28]: This is actually a development guide so I put `go run` here.

File: distribution/oceanbase-todo/README.md (hunk 1 above)
Comment (powerfooI) [reply distribution/oceanbase-todo/README.md:3]: OK

File: distribution/oceanbase-todo/ui/README.md (hunk 4 above)
Comment (powerfooI) [reply distribution/oceanbase-todo/ui/README.md:1]: Actually it's unnecessary to modify the generated file. The content in it is proper for the application.

**Review by chris-sun-star — APPROVED** [review]

**Pull request merged**